File: cvdl-rs/src/point.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Debug, Clone, Copy, PartialEq)]
pub struct Point {
    pub x: f32,
    pub y: f32,
}

impl Point {
    pub fn new(x: f32, y: f32) -> Self {
        Point { x, y }
    }

    pub fn move_x_by(&self, x: f32) -> Self {
        Point {
            x: self.x + x,
            y: self.y,
        }
    }

    pub fn move_y_by(&self, y: f32) -> Self {
        Point {
            x: self.x,
            y: self.y + y,
        }
    }
    #[allow(dead_code)]
    pub fn move_x_to(&self, x: f32) -> Self {
        Point { x, y: self.y }
    }

    pub fn move_y_to(&self, y: f32) -> Self {
        Point { x: self.x, y }
    }
}

File: cvdl-ts/src/Point.ts
export class Point {
	x: number;
	y: number;

	constructor(x: number, y: number) {
		this.x = x;
		this.y = y;
	}

	move_x_by(x: number): Point {
		return new Point(this.x + x, this.y);
	}

	move_y_by(y: number): Point {
		return new Point(this.x, this.y + y);
	}

	move_x_to(x: number): Point {
		return new Point(x, this.y);
	}

	move_y_to(y: number): Point {
		return new Point(this.x, y);
	}
}
